File: examples/canonical-faker/kotlin/src/main/kotlin/io/agentcontextbase/faker/domain/Entities.kt
package io.agentcontextbase.faker.domain

import com.fasterxml.jackson.annotation.JsonProperty

data class Organization(
    val id: String,
    val name: String,
    val slug: String,
    val plan: String,
    val region: String,
    @JsonProperty("created_at") val createdAt: String,
    @JsonProperty("owner_email") val ownerEmail: String,
)

data class User(
    val id: String,
    val email: String,
    @JsonProperty("full_name") val fullName: String,
    val locale: String,
    val timezone: String,
    @JsonProperty("created_at") val createdAt: String,
)

data class Membership(
    val id: String,
    @JsonProperty("org_id") val orgId: String,
    @JsonProperty("user_id") val userId: String,
    val role: String,
    @JsonProperty("joined_at") val joinedAt: String,
    @JsonProperty("invited_by") val invitedBy: String?,
)

data class Project(
    val id: String,
    @JsonProperty("org_id") val orgId: String,
    val name: String,
    val status: String,
    @JsonProperty("created_by") val createdBy: String,
    @JsonProperty("created_at") val createdAt: String,
)

data class AuditEvent(
    val id: String,
    @JsonProperty("org_id") val orgId: String,
    @JsonProperty("user_id") val userId: String,
    @JsonProperty("project_id") val projectId: String,
    val action: String,
    @JsonProperty("resource_type") val resourceType: String,
    @JsonProperty("resource_id") val resourceId: String,
    val ts: String,
)

data class ApiKey(
    val id: String,
    @JsonProperty("org_id") val orgId: String,
    @JsonProperty("created_by") val createdBy: String,
    val name: String,
    @JsonProperty("key_prefix") val keyPrefix: String,
    @JsonProperty("created_at") val createdAt: String,
    @JsonProperty("last_used_at") val lastUsedAt: String?,
)

data class Invitation(
    val id: String,
    @JsonProperty("org_id") val orgId: String,
    @JsonProperty("invited_email") val invitedEmail: String,
    val role: String,
    @JsonProperty("invited_by") val invitedBy: String,
    @JsonProperty("expires_at") val expiresAt: String,
    @JsonProperty("accepted_at") val acceptedAt: String?,
)

data class TenantCoreDataset(
    val profileName: String,
    val seed: Int,
    val organizations: List<Organization>,
    val users: List<User>,
    val memberships: List<Membership>,
    val projects: List<Project>,
    val auditEvents: List<AuditEvent>,
    val apiKeys: List<ApiKey>,
    val invitations: List<Invitation>,
)

File: examples/canonical-faker/php/src/Domain/Entities.php
<?php
declare(strict_types=1);

namespace AgentContextBase\Faker\Domain;

final readonly class Organization {
    public function __construct(
        public string $id,
        public string $name,
        public string $slug,
        public string $plan,
        public string $region,
        public string $created_at,
        public string $owner_email,
    ) {}
    public function toArray(): array { return get_object_vars($this); }
}

final readonly class User {
    public function __construct(
        public string $id,
        public string $email,
        public string $full_name,
        public string $locale,
        public string $timezone,
        public string $created_at,
    ) {}
    public function toArray(): array { return get_object_vars($this); }
}

final readonly class Membership {
    public function __construct(
        public string $id,
        public string $org_id,
        public string $user_id,
        public string $role,
        public string $joined_at,
        public ?string $invited_by,
    ) {}
    public function toArray(): array { return get_object_vars($this); }
}

final readonly class Project {
    public function __construct(
        public string $id,
        public string $org_id,
        public string $name,
        public string $status,
        public string $created_by,
        public string $created_at,
    ) {}
    public function toArray(): array { return get_object_vars($this); }
}

final readonly class AuditEvent {
    public function __construct(
        public string $id,
        public string $org_id,
        public string $user_id,
        public string $project_id,
        public string $action,
        public string $resource_type,
        public string $resource_id,
        public string $ts,
    ) {}
    public function toArray(): array { return get_object_vars($this); }
}

final readonly class ApiKey {
    public function __construct(
        public string $id,
        public string $org_id,
        public string $created_by,
        public string $name,
        public string $key_prefix,
        public string $created_at,
        public ?string $last_used_at,
    ) {}
    public function toArray(): array { return get_object_vars($this); }
}

final readonly class Invitation {
    public function __construct(
        public string $id,
        public string $org_id,
        public string $invited_email,
        public string $role,
        public string $invited_by,
        public string $expires_at,
        public ?string $accepted_at,
    ) {}
    public function toArray(): array { return get_object_vars($this); }
}

final readonly class Dataset {
    /**
     * @param Organization[] $organizations
     * @param User[] $users
     * @param Membership[] $memberships
     * @param Project[] $projects
     * @param AuditEvent[] $audit_events
     * @param ApiKey[] $api_keys
     * @param Invitation[] $invitations
     */
    public function __construct(
        public array $organizations,
        public array $users,
        public array $memberships,
        public array $projects,
        public array $audit_events,
        public array $api_keys,
        public array $invitations,
    ) {}
}
